src/commands/paranet/paranet-sync-command.js

```diff
@@ -423,19 +421,26 @@ class ParanetSyncCommand extends Command {
         paranetNodesAccessPolicy,
         operationId,
     ) {
-        const kasToSync = [];
-        for (let i = Number(startIndex); i <= contractKaCount; i += PARANET_SYNC_KA_COUNT) {
-            // Empty array, offset is 1 and we should probably start with zero
+        const pendingFilteredKAs = [];
+        let i = Number(startIndex);
+
+        const results = [];
+        while (i <= contractKaCount) {
             const nextKaArray =
                 await this.blockchainModuleManager.getParanetKnowledgeAssetsWithPagination(
                     blockchain,
                     paranetId,
                     i,
                     PARANET_SYNC_KA_COUNT,
                 );
-            if (!nextKaArray.length) break;
 
-            const filteredKAs = [];
+            // Break if no more KAs are fetched
+            if (!nextKaArray.length) {
+                break;
+            }
+
+            i += nextKaArray.length;
```

Comment:
Shouldn't this be ```nextKaArray.length + 1```, so in next interation you don't get last asset again as indexes start from 1?